data_structures/binary_tree/avl_tree.py

```diff
@@ -201,20 +207,52 @@ def del_node(root, data):
         return root
     if getheight(root.getright()) - getheight(root.getleft()) == 2:
         if getheight(root.getright().getright()) > getheight(root.getright().getleft()):
-            root = rightrotation(root)
+            root = leftrotation(root)
         else:
-            root = lrrotation(root)
+            root = rlrotation(root)
     elif getheight(root.getright()) - getheight(root.getleft()) == -2:
         if getheight(root.getleft().getleft()) > getheight(root.getleft().getright()):
-            root = leftrotation(root)
+            root = rightrotation(root)
         else:
-            root = rlrotation(root)
+            root = lrrotation(root)
     height = my_max(getheight(root.getright()), getheight(root.getleft())) + 1
     root.setheight(height)
     return root
 
 
 class AVLtree:
+    """
+    An AVL tree doctest
+    Examples:
+    >>> t = AVLtree()
+    >>> t.insert(4)
+    insert:4
+    >>> t.traversale()
```

Comment:
Let's change the method `traversale()` --> `__str__()` and then change instances of `t.traversale()` into `str(t)`.